File: Data Structures & Algorithms/binary-tree-diameter/submission-4.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    int ans;
    public int diameterOfBinaryTree(TreeNode root) {
        ans = 0;
        dia(root);
        return ans;
    }
    public int dia(TreeNode root) {
        if(root==null) return 0;
        int l = dia(root.left);
        int r = dia(root.right);
        ans = Math.max(ans, l+r);
        return 1+Math.max(l,r);
    }
}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-4.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        ans = []
        if not root:
            return ans
        q = deque()
        q.append(root)
        while q:
            ln = len(q)
            last = -1
            while ln>0:
                tmp = q.popleft()
                last = tmp.val
                if tmp.left:
                    q.append(tmp.left)
                if tmp.right:
                    q.append(tmp.right)
                ln -= 1
            ans.append(last)
        return ans